src/main/java/seedu/teachbook/model/student/Student.java

```diff
@@ -7,30 +7,43 @@
 import java.util.Objects;
 import java.util.Set;
 
+import seedu.teachbook.model.classobject.Class;
+import seedu.teachbook.model.classobject.ClassName;
 import seedu.teachbook.model.tag.Tag;
 
 /**
- * Represents a Person in the teachbook book.
+ * Represents a Student in TeachBook.
  * Guarantees: details are present and not null, field values are validated, immutable.
  */
 public class Student {
 
-    // Identity fields
+    // Compulsory fields
     private final Name name;
     private final Phone phone;
-    private final Email email;
+    private final Class studentClass;
 
-    // Data fields
+    // Optional fields
+    private final Email email;
     private final Address address;
     private final Set<Tag> tags = new HashSet<>();
 
     /**
      * Every field must be present and not null.
      */
     public Student(Name name, Phone phone, Email email, Address address, Set<Tag> tags) {
-        requireAllNonNull(name, phone, email, address, tags);
+        // TODO: add student to currently select class
+        // Current implementation: hardcoded
+        this(name, phone, new Class(new ClassName("A")), email, address, tags);
+    }
```

Comment:
If I understand correctly, this overloaded constructor should be removed finally, right?

Reply:
I did not think about this at first, but after some thought, yes I think it is better to not have two constructors in this case. For example, I think it is more consistent to get the currently selected class in the command parsers, ie. `AddCommandParser`, as per how all other fields, `name`, `phone`, etc., are dealt with. However, there are quite a few places that require changes. I think it is better to change them now, right?

Reply:
Keep the overloaded constructor for now, thus no need to change other places. Also consider whether the class should be final or not, and provide a `setClass()` method? Actually I don't think letting command parse know the currently selected class is a good way, but it is possible to let class be `null` first and set it later in model component?

Reply:
Hm why should `studentClass` not be `final`? I think it is alright for `studentClass` to be `final` since it is immutable. Even if for `edit`, a new `Student` will be created to replace the existing one, as per the current implementation. And why getting the currently selected class in the parser is not a good way? Anyway, I think I just leave the current code as it is for now.

Reply:
haha sure if you prefer everything to be immutable.